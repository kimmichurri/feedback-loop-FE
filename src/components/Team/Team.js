import React, { Component } from 'react'

export class Team extends Component {

  render() {
    // let displayMembers
    // console.log(this.props.teamMembers)
    // if(this.props.teamMembers.length) {
    //   displayMembers = this.props.teamMembers.map(member => {
    //     return <p key={member.id}>{member.name}</p>
    //   })
    // }
    
    const names = this.props.members.map(member => {
      return member.name
    })
    return(
      <div className='team' id={this.props.id}>
        Team
        {names}
      </div>
    )
  } 
}

export default Team