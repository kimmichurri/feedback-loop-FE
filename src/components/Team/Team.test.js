import React from 'react'
import { shallow } from 'enzyme'
import Team from './Team'

describe('Team', () => {
  let wrapper
  let mockTeam = {
    id: 4,
    members: [{id: 1, name: "Taylor"}, {id: 2, name: "Kim"}]
  }
  let mockFn = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<Team 
      id={mockTeam.id}
      members={mockTeam.members}
      handleTeamName={mockFn}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should update state onChange', () => {
    const mockEvent = { target: { name: 'name', value: 'xyz'}}
    wrapper.find('.team-input').simulate('change', mockEvent)
    expect(wrapper.state('name')).toBe('xyz')
  })
})