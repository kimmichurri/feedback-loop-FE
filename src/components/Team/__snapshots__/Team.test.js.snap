// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team should match the snapshot 1`] = `
<div
  className="team"
  id={4}
>
  <input
    autoComplete="off"
    className="team-input"
    name="name"
    onChange={[Function]}
    placeholder="Name this team"
  />
  <div
    className="assigned-student"
    key="1"
  >
    Taylor
  </div>
  <div
    className="assigned-student"
    key="2"
  >
    Kim
  </div>
</div>
`;
